src/main/java/taxiapp/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,58 @@
+package taxiapp.dao;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import taxiapp.db.Storage;
+import taxiapp.lib.Dao;
+import taxiapp.model.Car;
+import taxiapp.model.Driver;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+
+    @Override
+    public Car create(Car car) {
+        Storage.addCar(car);
+        return Storage.cars.get(Storage.cars.size() - 1);
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        return Storage.cars.stream()
+                .filter(c -> c.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return Storage.cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        Car oldValue = get(car.getId()).orElseThrow();
+        Storage.cars.remove(oldValue);
+        Storage.cars.add(car);
+        return oldValue;
+    }
```

Comment:
```suggestion
    public Car update(Car car) {
        IntStream.range(0, Storage.cars.size())
                .filter(i -> Storage.cars.get(i).getId().equals(car.getId()))
                .forEach(i -> Storage.cars.set(i, car));
        return car;
    }
```
let's replace each occurrence of the car with the same id in Storage - with the new car